qiskit/algorithms/amplitude_amplifiers/amplification_problem.py

```diff
@@ -67,8 +64,6 @@ def __init__(
             self._is_good_state = is_good_state
         elif hasattr(oracle, "evaluate_bitstring"):
             self._is_good_state = oracle.evaluate_bitstring
-        else:
-            raise TypeError("A is_good_state function is required with the provided oracle")
```

Comment:
Removing this causes `self._is_good_state` to be undefined if `is_good_state=None` and `oracle` doesn't have the `evalute_bitstring` attribute, and then `self.is_good_state` will raise `AttributeError`.